espflash/src/cli/config.rs

```diff
@@ -75,91 +75,133 @@ impl UsbDevice {
 /// Deserialized contents of a configuration file
 #[derive(Debug, Deserialize, Serialize, Default, Clone)]
 pub struct Config {
+    pub project_config: ProjectConfig,
+    pub port_config: PortConfig,
+}
+
+/// Deserialized contents of a configuration file
+#[derive(Debug, Deserialize, Serialize, Default, Clone)]
+pub struct ProjectConfig {
     /// Baudrate
     #[serde(default)]
     pub baudrate: Option<u32>,
     /// Bootloader path
     #[serde(default)]
     pub bootloader: Option<PathBuf>,
-    /// Preferred serial port connection information
-    #[serde(default)]
-    pub connection: Connection,
     /// Partition table path
     #[serde(default)]
     pub partition_table: Option<PathBuf>,
     /// Partition table offset
     #[serde(default)]
     pub partition_table_offset: Option<u32>,
-    /// Preferred USB devices
-    #[serde(default)]
-    pub usb_device: Vec<UsbDevice>,
     /// Flash settings
     #[serde(default)]
     pub flash: FlashSettings,
+}
+
+/// Deserialized contents of a serial port configuration file
+#[derive(Debug, Deserialize, Serialize, Default, Clone)]
+pub struct PortConfig {
+    /// Preferred serial port connection information
+    #[serde(default)]
+    pub connection: Connection,
+    /// Preferred USB devices
+    #[serde(default)]
+    pub usb_device: Vec<UsbDevice>,
     /// Path of the file to save the configuration to
     #[serde(skip)]
     save_path: PathBuf,
```

Comment:
Only PortConfig requires `save_path`, as its the only config file that can be saved, `save_with` is only called here: https://github.com/esp-rs/espflash/blob/main/espflash/src/cli/serial.rs#L53